--- cc/private/toolchain/windows_cc_toolchain_config.bzl
+++ cc/private/toolchain/windows_cc_toolchain_config.bzl
@@ -96,7 +96,7 @@
 ]
 
 def _use_msvc_toolchain(ctx):
-    return ctx.attr.cpu in ["x64_windows", "arm64_windows"] and (ctx.attr.compiler == "msvc-cl" or ctx.attr.compiler == "clang-cl")
+    return ctx.attr.cpu in ["x64_windows", "x64_x86_windows", "arm64_windows"] and (ctx.attr.compiler == "msvc-cl" or ctx.attr.compiler == "clang-cl")
 
 def _impl(ctx):
     if _use_msvc_toolchain(ctx):
@@ -813,7 +813,7 @@
                     ],
                     flag_groups = [
                         flag_group(
-                            flags = default_compile_flags_list,
+                            flags = default_compile_flags_list + ctx.attr.default_compile_flags,
                         ),
                     ],
                 ),
